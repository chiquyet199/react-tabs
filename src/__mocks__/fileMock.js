export function MockFile() {}

MockFile.prototype.create = (
  name = "mock.jpg",
  size = 1024,
  mimeType = "image/png",
) => {
  function range(count) {
    let output = "";
    for (let i = 0; i < count; i += 1) {
      output += "a";
    }

    return output;
  }

  const blob = new Blob([range(size)], {type: mimeType});
  blob.lastModifiedDate = new Date();
  blob.name = name;

  return blob;
};
