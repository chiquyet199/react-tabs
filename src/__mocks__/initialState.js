export const initialState = {
  locale: {
    currentLocale: "en-US",
    translations: {
      common: {
        postcode_lookup: {
          postcode: "Postcode",
          find_address: "Find address",
          address_not_listed: "Address not listed",
        },
      },
      home__welcome: "Welcome to RNW starter app",
      home__loggedin: "You are now logged in",
      login__email__label: "Email address",
      login__password__label: "Password",
      login__email__placeholder: "example@example.com",
      login__password__placeholder: "******",
      login__continue__with__facebook: "Continue with Facebook",
      login: "Sign in",
      create__account: "Create account",
      login__forgot__password: "Forgot password",
      logout: "Logout",
      back: "Back",
      submit: "Submit",
      rent_passport: "Rent Passport",
      deposit_free: "Deposit Free",
      home_management: "Home Management",
      authorize: "Authorize",
      terms_header: "Canopy terms & conditions",
      terms:
        "1. INTRODUCTION\n\nBy signing up for an account with Canopy, you are agreeing to be bound by the following terms and conditions (“Terms of Service”). These Terms of Service are to be read in conjunction with:\n\nour Privacy Policy, which sets out the terms on which we process any personal data we collect from you, or that you or others provide to us; and\n\nour Cookies Policy, which sets out information about the cookies we use.\n\nThese Terms of Service are important. Please read them carefully as they contain details of our statutory and regulatory responsibilities to you, and your contractual obligations to us. They also set out the basis on which we provide, and you are permitted to use, the various services made available on the website www.findyourcanopy.com and the mobile application known as ‘Canopy’ (together, the “Platform”), including the following:",
      verify__email__title__text: "Verify your email address",
      verify__email__text:
        "We need to know you are you before creating a RentPassport. Please click the link in the email we sent you.",
      open__email__client: "Open my email client",
      resend__verification__email: "Resend verification email",
      not__received__an__email: "Not received an email?",
      forgot_password: "Forgot your password?",
      reset_password: "Change your password",
      tenant_reference_submitted: "Reference submitted",
      tenant_reference_thanks_for_time:
        "Thanks for taking the time to submit a tenant reference for",
      tenant_reference_did_you_know:
        "Did you know that Canopy offers the following property management tools to tenants, landlords and agents:",
      tenant_reference_rent_passport_text:
        "Full tenant referencing and property management including deposits and payments",
      tenant_reference_deposit_free_text:
        "Qualified tenants can pay a premium to cover their rental security deposit",
      tenant_reference_home_management_text:
        "With a valid UK Open Banking account, tenants are able to have rent payments contribute to their Experian credit report",
      find_out_more: "Find out more",
      tenant_reference_title: "Tenant reference request",
      upload_tenant_reference_document: "Upload tenant reference document",
      upload_tenant_reference_document_description:
        "If you have a manual form you usually use, please upload it here.",
      upload: "Upload",
      submit_reference: "Submit reference",
      tenant_reference_request: "Tenant reference request",
      complete_reference_form_description:
        " tenant of yours, has authorised us to request a reference from you.\n" +
        "As you know a landlord reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
      upload_reference_document: "Upload reference document",
      complete_reference_form: "Complete reference form",
      complete_reference_form_a: ", a ",
      truelayer: {
        truelayer_auth_title: "Truelayer Authorize",
      },
      employer_reference_form_description:
        " has advised us that you are their employer and has authorised us to request a reference from you.\n" +
        "As you know an employment reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
    },
  },
  auth: {
    user: null,
    isLoggedIn: false,
  },
  dimensions: {
    fontScale: 1,
    height: 901,
    scale: 1,
    width: 1920,
    size: "L",
  },
  inputField: {
    showPassword: true,
  },
  form: {},
  partialForm: {
    values: {},
  },
  progress: 0,
  notification: {
    type: null,
    message: "",
  },
  user: {
    profile: {
      firstName: "FirstName",
      lastName: "LastName",
    },
  },
  router: {
    location: {
      pathname: "/tenant-reference-request/complete",
      search: "",
      hash: "",
    },
    action: "POP",
  },
  lookups: {
    addresses: [],
  },
  topbar: {
    haveTopbar: true,
    topbarBtn: {
      haveTopbarButton: false,
      topbarButtonText: "",
      btnType: "transparent_green",
      topbarButtonIcon: undefined,
      hideBackButton: false,
    },
    topbarCenter: {
      topbarCenterText: "Center",
    },
    topbarLeft: {
      topbarLeftText: "Back",
    },
  },
  images: {
    accepted: [],
  },
  rentPassport: {
    aboutYou: {
      email: "jonathan@gmail.com",
      firstName: "Jonathan",
      lastName: "Beaumont",
      middleName: null,
      gender: "MALE",
      dayOfBirth: 20,
      monthOfBirth: 11,
      yearOfBirth: 1970,
      maritalStatus: "MARRIED",
      phoneNumber: "01123456789",
      nationality: "UK",
      isSmoker: true,
      hasPets: true,
      hasChildren: true,
      status: "GREEN",
      messages: [],
    },
    financial: {
      salary: 100000,
      pension: 0,
      interest: 0,
      benefits: 0,
      rentals: 0,
      other: 0,
      grossAnnualIncome: 100000,
      sourcesOfIncome: ["Employment"],
      status: "YELLOW",
      messages: [],
      bankrupt: false,
      creditScore: 930,
      depositFree: true,
      maximumEligible: 1800.266,
      activeCCJs: 0,
      activeCCJYear: 0,
      activeCCJMonth: 0,
      activeCCJValue: 0,
      previousCCJs: 0,
      previousCCJsYear: 0,
      previousCCJsMonth: 0,
      activeInsolvencies: 0,
      activeInsolvenciesYear: 0,
      activeInsolvenciesMonth: 0,
      previousInsolvencies: false,
      previousInsolvenciesYear: 0,
      previousInsolvenciesMonth: 0,
    },
    rentPassportLegal: {
      pep: false,
      mortality: "passed",
      legalWarnings: false,
      sanctions: false,
      status: "GREEN",
      messages: [],
    },
    whatYouDo: {
      workInfo: [
        {
          workStatus: "EMPLOYED",
          companyName: "companyName",
          jobTitle: "jobTitle",
          income: 100000,
          employmentTime: "FULL_TIME",
          employmentType: "CONTRACT",
          workedSinceMonth: 12,
          workedSinceYear: 2017,
          studentType: "POST_GRADUATE_STUDENT",
          reference: {
            contactFirstName: "Silvester",
            contactLastName: "Stallone",
            contactEmailAddress: "rambo666@gmail.com",
            referredAt: "12-12-2016",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
    whereYouLived: {
      residenceInfo: [
        {
          line1: "Addresss line 1",
          town: "Town",
          state: "State",
          postCode: "UL11223",
          countryCode: "UK",
          residenceType: "RENTING_FROM_AGENT",
          livedSinceMonth: 1,
          livedSinceYear: 2016,
          livedToMonth: null,
          livedToYear: null,
          reference: {
            contactFirstName: "Jonathan",
            contactLastName: "Beaumont",
            contactEmailAddress: "jonathan@gmail.com",
            referredAt: "01-01-2018",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
  },
};
