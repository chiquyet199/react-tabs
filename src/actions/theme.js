import {UPDATE_GLOBAL_THEME} from "./types";

export const updateTheme = globaltheme => ({
  type: UPDATE_GLOBAL_THEME,
  globaltheme,
});
