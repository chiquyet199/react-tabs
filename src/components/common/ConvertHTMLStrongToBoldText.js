import React from "react";
import {Text} from "react-native";
import styled from "styled-components/native";

export const StyledView = styled(Text)`
  display: inline;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export default ({htmlString}) => {
  const parsed = [];
  let cursor = 0;
  let searchingFor = "<strong>";
  while (cursor < htmlString.length) {
    let endPoint = htmlString.indexOf(searchingFor, cursor);
    endPoint = endPoint === -1 ? htmlString.length : endPoint;
    const token = htmlString.substring(cursor, endPoint);
    if (searchingFor === "<strong>") {
      parsed.push(<Text key={token}>{token}</Text>);
    } else {
      // Styled-components seems to ignore font-weight here
      parsed.push(
        <Text style={{fontWeight: "bold"}} key={token}>
          {token}
        </Text>,
      );
    }
    cursor = endPoint + searchingFor.length;
    searchingFor = searchingFor === "<strong>" ? "</strong>" : "<strong>";
  }

  return <StyledView>{parsed}</StyledView>;
};
