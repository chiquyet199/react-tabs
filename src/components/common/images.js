import styled from "styled-components/native";

export const CenterIconImage = styled.Image`
  width: 80px;
  height: 80px;
  margin-top: 20px;
  margin-bottom: 20px;
  align-self: center;
`;

export const CenterPageImage = styled.Image`
  width: 370px;
  height: 320px;
  margin-top: 20px;
`;
