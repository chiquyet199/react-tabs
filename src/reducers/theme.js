import {UPDATE_GLOBAL_THEME} from "../actions/types";

const initialState = "light";

export default (state = initialState, {type, globaltheme}) => {
  switch (type) {
    case UPDATE_GLOBAL_THEME:
      return globaltheme;
    default:
      return state;
  }
};
