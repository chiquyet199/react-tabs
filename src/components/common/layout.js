import React from "react";
import styled from "styled-components/native";
import {isNative, isWeb, responsive, themed} from "../../utils/common";
import {TouchableOpacity} from "react-native";
import {DESKTOP_CONTAINER_WIDTH} from "../../constants/sizes";
import {ButtonContainer} from "../Button/Button";
import {func, node, number, string} from "prop-types";

export const FlexContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
`;

export const Flexed = styled.View``;

export const Flex = styled.View`
  margin-top: ${({theme}) => theme.padding.top + "px"};
  position: absolute;
  width: 100%;
  background-color: transparent;
`;

export const FlexColumn = styled.View`
  max-width: 537px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const TopBarContainer = styled.View`
  width: 100%;
  max-width: 100%;
  padding: 0;
  padding-top: ${({theme}) => theme.padding.top + "px"}
  z-index: 2;
  position: relative;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].topbar.background}
`;

export const BaseContainer = styled.View`
  flex: 1;
  padding: 0;
  width: 100%;
  z-index: 1;
`;

export const CenteredContainer = styled.View`
  flex: 1;
  justify-content: ${({align}) => align || "center"};
  align-self: center;
  align-items: center;
  width: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
  margin-horizontal: auto;
  padding-horizontal: ${({full}) => (full === "true" ? "0" : "10px")};
  flex-basis: auto;
`;

export const EndPageContainer = responsive(CenteredContainer)({
  S: `justify-content: flex-start`,
});

export const PageContainer = responsive(styled(CenteredContainer)`
  width: 100%;
  max-width: ${({full}) =>
    full === "true" ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  align-self: center;
  align-items: stretch;
  margin-top: ${({layout}) => (layout === "hq" ? "60px" : "40px")};
  ${({margintop}) => margintop >= 0 && `margin-top: ${margintop}px`};
  padding-horizontal: 10px;
  justify-content: center;
  flex-basis: auto;
`)({
  S: `
      padding-top: ${({theme}) => theme.padding.top + "px"};
      justify-content: space-between;
      padding-horizontal: 10px;
      margin-top: ${({layout}) => (layout === "hq" ? "80px" : "20px")};
    `,
});

export const FormContainer = responsive(styled.View`
  justify-content: center;
  align-items: center;
  max-width: ${DESKTOP_CONTAINER_WIDTH}px;
  width: 100%;
  align-self: center;
`)({
  S: `
      padding-top: 10px;
      justify-content: space-between;
      max-width: 100%;
    `,
});

export const FlexRow = styled.View`
  flex: 1;
  flex-direction: row;
  flex-basis: 100%;
`;

export const ButtonBar = styled.View`
  max-width: ${({isMenu}) =>
    isMenu ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  width: 100%;
  border-top-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.lightSteel : theme.colors.none};
  border-top-width: ${({isMenu}) => (isMenu ? "1px" : 0)};
  background-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.white : theme.colors.none};
  flex-direction: row;
  margin-horizontal: 0;
  margin-bottom: ${({isMenu}) => (isMenu ? 0 : "5px")};
  flex-wrap: ${({wrap}) => (wrap ? "wrap" : "nowrap")};
  ${({mt}) => mt && `margin-top: ${mt}px}`};
  ${ButtonContainer} {
    margin: 20px;
  }
`;

export const BottomButtonBar = responsive(ButtonBar)({
  S: `flexGrow: 10;
      flexDirection: row;
      alignItems: center;
      justifyContent: flex-end;
    `,
});

export const ResponsiveWebScrollView = responsive(styled.ScrollView``)({
  M: `
        width: 500px;
  `,
  L: `
        width: 600px;
    `,
});

export const PageTitle = themed(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding-horizontal: 5px;
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  ${({mb}) => mb && `margin-bottom: ${mb}px}`};
  ${({color}) => color && `color: ${color}`};
`);

export const TextParagraph = themed(styled.Text`
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  color: ${({disabled, theme, globaltheme}) =>
    disabled
      ? theme[globaltheme].disabledTextColor
      : theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding: ${({gutters}) => gutters || "15px 5px"};
  flex-shrink: 0;
  flex-grow: 0;
  ${({mb}) => mb && `margin-bottom: ${mb}px`};
  ${({color}) => color && `color: ${color}`};
`);

export const ViewParagraph = themed(styled.View`
  width: 100%;
  flex-shrink: 0;
  flex-grow: 0;
`);

export const SubTitle = themed(styled(TextParagraph)`
  font-weight: bold;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  ${({color}) => color && `color: ${color}`};
`);

export const TextLinkText = themed(styled.Text`
  text-align: center;
  font-size: 16;
  line-height: 24;
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: 700;
  padding-top: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")};
  padding-bottom: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-left: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-right: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  text-align:  ${({align}) => align || "center"};
`);

export const TextLink = ({onClick, children, gutters, align, globaltheme}) => (
  <TouchableOpacity onPress={onClick}>
    <TextLinkText align={align} gutters={gutters} globaltheme={globaltheme}>
      {children}
    </TextLinkText>
  </TouchableOpacity>
);

TextLink.propTypes = {
  onClick: func,
  children: node,
  gutters: number,
  align: string,
  globaltheme: string,
};

export const NotchSafety = styled.View`
  height: 40px;
`;

export const PageHeader = styled.Text`
  font-size: 18;
  text-align: center;
  width: 100%;
  margin-top: ${({theme}) => theme.padding.top + 15 + "px"};
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  position: absolute;
`;

export const HeadlineText = responsive(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  margin: 8px 0 40px;
  padding: 0;
`)({
  S: `
    padding-left: 15px;
    `,
  L: `
    align-self: center;
   `,
});

export const BoxButtonBar = styled.View`
  align-self: center;
  align-items: stretch;
  width: 100%;
  margin-horizontal: auto;
  flex-direction: row;
`;

export const FixedWidthContainer = styled.View`
  width: ${DESKTOP_CONTAINER_WIDTH}px;
  margin-horizontal: auto;
  padding-horizontal: 15px;
`;

export const PaddedContent = styled.View`
  padding-horizontal: 10px;
`;

export const ErrorMessage = themed(styled(TextParagraph)`
  color: ${({theme}) => theme.colors.alertRed};
`);

// Scroll layout

export const ScrollContentContainer = styled.View`
  width: 100%;
  ${isNative ? "flex: 1;" : "height: calc(100vh - 52px)"};
  flex-grow: ${() => (isWeb ? 0 : 1)};
  flex-shrink: ${() => (isWeb ? 0 : 1)};
  justify-content: center;
  align-self: center;
  align-items: center;
`;

export const ScrollButtonBar = styled(ButtonBar)`
  padding-horizontal: 10px;
  ${({mt}) => mt && `margin-top: ${mt}px}`};
`;

export const ScrollScrollView = styled.ScrollView`
  ${isNative ? "" : "height: 100%"};
  width: 100%;
  flex-grow: 1;
  flex-shrink: 1;
`;
