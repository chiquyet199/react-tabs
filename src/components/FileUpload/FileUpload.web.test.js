import React from "react";
import {shallow} from "enzyme";
import {
  FileUpload,
  StyledButton,
  StyledIcon,
  UploadedFile,
} from "./WebFileUpload";
import {Field} from "redux-form";
import configureStore from "redux-mock-store";
import theme from "../../../../canopy-hq/src/constants/theme";
import {initialState} from "../../__mocks__/initialState";
const mockStore = configureStore();
import {extend} from "underscore";
import {MockFile} from "../../__mocks__/fileMock.js";
import {isWeb} from "../../utils/common";

describe("WebFileUpload tests", () => {
  it("matches snapshot", () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("use `fileUpload` key in schema if no fieldName is provided", () => {
    const component = getComponent().dive();
    expect(component.find(Field).prop("name")).toBe("fileUpload");
  });

  it("includes dynamic fieldName in schema as a key", () => {
    const fieldName = "customName";
    const component = getComponent({fieldName});
    expect(component.find(Field).prop("name")).toBe(fieldName);
  });

  it("renders Dropzone component and renders list then file is passed and removes on click", async () => {
    const fieldName = "customName";
    const reset = jest.fn();
    const field = {
      input: {
        onChange: jest.fn(),
      },
      meta: {
        error: {},
        touched: false,
      },
    };
    const component = getComponent({fieldName, reset});
    const mock = new MockFile();
    const file = isWeb ? await mock.create() : getMockFileNative();
    component.setState({accepted: [file]});
    component.update();
    expect(component).toMatchSnapshot();
    const dropzone = component.find(Field).prop("component")(field);
    const dropzoneComp = shallow(dropzone);
    expect(dropzoneComp.prop("accepted")).toBe(1);
    expect(dropzoneComp.find(StyledButton).prop("middle")).toBe(
      mockProps.btnText,
    );
    expect(dropzoneComp.find(StyledButton).prop("type")).toBe(
      mockProps.btnType,
    );
    component.find(StyledIcon).prop("onPress")();
    expect(reset).toHaveBeenCalled();
    component.update();
    expect(component.find(UploadedFile)).not.toExist();
  });

  const mockProps = {
    btnText: "Select file",
    btnType: "secondary",
    onSubmit: () => {},
    globaltheme: "light",
    theme,
    reset: () => {},
  };
  const getComponent = props => {
    const parsedProps = extend(mockProps, props);

    return shallow(<FileUpload {...parsedProps} />, {
      context: {store: mockStore(initialState)},
    });
  };

  const getMockFileNative = () => ({
    size: 1024,
    type: "image/png",
    name: "File Name",
  });
});
