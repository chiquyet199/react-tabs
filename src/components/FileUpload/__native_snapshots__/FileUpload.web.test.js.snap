// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebFileUpload tests matches snapshot 1`] = `
<styled.div>
  <aside>
    <styled.ul />
  </aside>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      name="fileUpload"
    />
  </form>
</styled.div>
`;

exports[`WebFileUpload tests renders Dropzone component and renders list then file is passed and removes on click 1`] = `
<styled.div>
  <aside>
    <styled.ul>
      <styled.li
        key="File Name"
      >
        <Icon
          allowFontScaling={false}
          color="#6b7c93"
          name="legal"
          size={32}
        />
        <styled.span>
          File Name
        </styled.span>
        <Styled(Icon)
          globaltheme="light"
          name="close"
          onPress={[Function]}
          size={32}
          theme={
            Object {
              "buttons": Object {
                "danger": Object {
                  "background": "#f16651",
                  "border": "#f16651",
                  "fontWeight": "bold",
                  "text": "#ffffff",
                },
                "facebook": Object {
                  "background": undefined,
                  "border": undefined,
                  "fontWeight": "bold",
                  "text": "#ffffff",
                },
                "primary": Object {
                  "background": "#5ab1b8",
                  "border": "#5ab1b8",
                  "fontWeight": "bold",
                  "text": "#ffffff",
                },
                "primary_outlined": Object {
                  "background": "rgba(0, 0, 0, 0)",
                  "border": "#5ab1b8",
                  "fontWeight": "bold",
                  "text": "#5ab1b8",
                },
                "secondary": Object {
                  "background": "#ffffff",
                  "border": "#5ab1b8",
                  "fontWeight": "bold",
                  "text": "#5ab1b8",
                },
                "site_page_util_bar_primary": Object {
                  "background": "#5ab1b8",
                  "border": "#5ab1b8",
                  "fontSize": "14px",
                  "fontWeight": "bold",
                  "text": "#ffffff",
                },
                "tabButton": Object {
                  "background": "#ffffff",
                  "border": "#e3eaf2",
                  "fontSize": "14px",
                  "fontWeight": "bold",
                  "text": "#6b7c93",
                },
                "tertiary": Object {
                  "background": "#ffffff",
                  "border": undefined,
                  "fontWeight": "bold",
                  "text": undefined,
                },
                "transparent": Object {
                  "background": "rgba(0, 0, 0, 0)",
                  "border": "rgba(0, 0, 0, 0)",
                  "fontWeight": "bold",
                  "text": "#ffffff",
                },
                "transparent_gray": Object {
                  "background": "rgba(0, 0, 0, 0)",
                  "border": "rgba(0, 0, 0, 0)",
                  "fontWeight": "bold",
                  "text": "#6b7c93",
                },
                "transparent_green": Object {
                  "background": "rgba(0, 0, 0, 0)",
                  "border": "rgba(0, 0, 0, 0)",
                  "fontSize": "14px",
                  "fontWeight": "bold",
                  "text": "#5ab1b8",
                },
              },
              "colors": Object {
                "alertRed": "#f16651",
                "black": "#000000",
                "canopyGreen": "#5ab1b8",
                "canopySoftSteel": "#8999ab",
                "canopySoftSteelOverlay": "#f3f7fc",
                "canopySteel": "#6b7c93",
                "darkMint": "#5dc471",
                "darkOverlay": "rgba(0, 0, 0, 0.8)",
                "darkTeal": "#5ab1b8",
                "darkTransparent": "rgba(0, 0, 0, 0.7)",
                "lightSteel": "#e3eaf2",
                "lightTeal": "#37b3ba",
                "midTeal": "#42c8ce",
                "none": "rgba(0, 0, 0, 0)",
                "primary": "#5ab1b8",
                "shadowColor": "rgba(107, 124, 147, 0.4)",
                "steel": "#b9c7d6",
                "switchColor": "#5dc471",
                "veryLightSteel": "#f6f9fc",
                "white": "#ffffff",
              },
              "dark": Object {
                "accentColor": "#ffffff",
                "background": Array [
                  undefined,
                  undefined,
                ],
                "baseColor": "#ffffff",
                "form": Object {
                  "accentColor": "#5ab1b8",
                  "baseColor": "#ffffff",
                  "bounceButton": "transparent",
                },
                "switch": Object {
                  "accentColor": "#5dc471",
                  "baseColor": "#ffffff",
                },
                "textColor": "#ffffff",
              },
              "hq": Object {
                "accentColor": "#5ab1b8",
                "background": Array [
                  "#ffffff",
                  "#ffffff",
                ],
                "baseColor": "#6b7c93",
                "form": Object {
                  "accentColor": "#5ab1b8",
                  "baseColor": undefined,
                },
                "switch": Object {
                  "accentColor": "#5dc471",
                  "baseColor": "#ffffff",
                },
                "textColor": "#6b7c93",
              },
              "light": Object {
                "accentColor": "#5ab1b8",
                "background": Array [
                  "#ffffff",
                  "#ffffff",
                ],
                "baseColor": "#6b7c93",
                "form": Object {
                  "accentColor": "#5ab1b8",
                  "baseColor": "#6b7c93",
                  "bounceButton": "transparent_green",
                },
                "switch": Object {
                  "accentColor": "#5dc471",
                  "baseColor": "#ffffff",
                },
                "textColor": "#6b7c93",
              },
              "padding": Object {
                "top": 0,
              },
            }
          }
        />
      </styled.li>
    </styled.ul>
  </aside>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      name="customName"
    />
  </form>
</styled.div>
`;
